### PR TITLE
feat: api management

### PR DESCRIPTION
**Pull request by jspdown**
### What does this PR do?

This PR adds API management CRDs:
- APIPortal
- APIGateway
- APIAccess
- APICollection
- API

And update the ACP CRD with the new apiKey and oauthIntrospection.

### Motivation

Make these features available on newly installed agents.

### More

- [x] Yes, I updated the tests accordingly
- [x] Yes, I ran `make test` and all the tests passed

### CODE REVIEW

File: hub-agent/templates/rbac/cluster-role-dev-portal.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+metadata:
+  name: hub-agent-dev-portal
+  labels:
+    app.kubernetes.io/name: {{ template "hub-helm-chart.name" . }}
+    helm.sh/chart: {{ template "hub-helm-chart.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/version: {{ .Chart.Version }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+rules:
+  # Access Control RBACs.
```
Comment (lbenguigui) [inline hub-agent/templates/rbac/cluster-role-dev-portal.yaml]:
```suggestion
```

File: hub-agent/templates/rbac/service-accounts.yaml [hunk 2]
```diff
@@ -37,3 +37,22 @@ metadata:
...
+kind: ServiceAccount
+metadata:
+  name: hub-agent-dev-portal
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/name: {{ template "hub-helm-chart.name" . }}
+    helm.sh/chart: {{ template "hub-helm-chart.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/version: {{ .Chart.Version }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
```
Comment (tomMoulard) [inline hub-agent/templates/rbac/service-accounts.yaml]:
```suggestion
    {{- include "hub-helm-chart.labels" . | nindent 4 }}
```

File: hub-agent/templates/services.yaml [hunk 3]
```diff
@@ -46,3 +46,30 @@ spec:
...
+kind: Service
+metadata:
+  name: hub-agent-dev-portal
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/name: {{ template "hub-helm-chart.name" . }}
+    helm.sh/chart: {{ template "hub-helm-chart.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/version: {{ .Chart.Version }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
```
Comment (tomMoulard) [inline hub-agent/templates/services.yaml]:
```suggestion
    {{- include "hub-helm-chart.labels" . | nindent 4 }}
```

**Review by mmatur — APPROVED** [review]
LGTM

File: hub-agent/templates/deployment-dev-portal.yaml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+              port: 80
+            initialDelaySeconds: 5
+            periodSeconds: 20
+          resources:
+            {{- with .Values.resources }}
+            {{- toYaml . | nindent 12 }}
+            {{- end }}
+          securityContext:
+            allowPrivilegeEscalation: false
+            readOnlyRootFilesystem: false
```
Comment (mloiseleur) [inline hub-agent/templates/deployment-dev-portal.yaml:63]: If possible, I would suggest to restrict written data to a specific path.
(And so sets readOnlyRootFilesystem to true with an emptyDir mounted for temporary data)

**Review by mloiseleur — APPROVED** [review]
LGTM

**Pull request merged**